src/test/javascript/spec/interceptor/auth-expired.interceptor.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { AuthExpiredInterceptor } from 'app/core/interceptor/auth-expired.interceptor';
+import { LoginService } from 'app/core/login/login.service';
+import { StateStorageService } from 'app/core/auth/state-storage.service';
+import { Router } from '@angular/router';
+import { AuthServerProvider } from 'app/core/auth/auth-jwt.service';
+import { throwError } from 'rxjs';
+
+describe(`AuthExpiredInterceptor`, () => {
+    let authInterceptor: AuthExpiredInterceptor;
+
+    let loginServiceMock: LoginService;
+    let stateStorageServiceMock: StateStorageService;
+    let authServerProviderMock: AuthServerProvider;
+
+    const routerMock = {
+        routerState: {
+            snapshot: {
+                url: 'https://example.com',
+            },
+        },
+    } as any as Router;
+
+    beforeEach(() => {
+        loginServiceMock = {
+            logout: jest.fn(),
+        } as any as LoginService;
+        stateStorageServiceMock = {
+            storeUrl: jest.fn(),
+        } as any as StateStorageService;
+        authServerProviderMock = {
+            getToken: jest.fn(),
+        } as any as AuthServerProvider;
+
+        authInterceptor = new AuthExpiredInterceptor(loginServiceMock, stateStorageServiceMock, routerMock, authServerProviderMock);
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should logout and store the current url if status is 401 and there is no token', () => {
+        const mockHandler = {
+            handle: () => throwError(() => new HttpErrorResponse({ status: 401 })),
+        };
+        const tokenSpy = jest.spyOn(authServerProviderMock, 'getToken').mockReturnValue('token-123');
+
+        authInterceptor.intercept(null as any, mockHandler).subscribe();
```

Comment:
Please use an actual value. Same for other occurrences

Reply:
👍